R/expect-setequal.R

```diff
@@ -78,9 +78,20 @@ expect_mapequal <- function(object, expected) {
     abort("`object` and `expected` must both be vectors")
   }
 
+  # Length-0 vectors are OK whether named or unnamed.
+  if (length(act$val) == 0 && length(exp$val) == 0) {
+    warn("`object` and `expected` are empty lists")
+    succeed()
+    return(invisible(act$val))
+  }
+
   act_nms <- names(act$val)
   exp_nms <- names(exp$val)
 
+  if (!names_ok(act_nms, "object") || !names_ok(exp_nms, "expected")) {
+    return(act$val)
```

Comment:
Shouldn't this be a failure or error?

Reply:
Oh you put the `fail` in `names_ok()` — I think you need to bring it back into this function

Reply:
Rename `names_ok()` to `check_names_ok()` and have it call `stop()` rather than `fail()`. Then you can discard this `if` block.